### PR TITLE
File type classifier values

### PR DESCRIPTION
**Pull request by kadmit**
Task: https://github.com/keeleinstituut/tv-tolkevarav/issues/88

Implemented migration for inserting file type classifier values.

### CODE REVIEW

File: application/database/migrations/2023_04_03_122625_add_file_type_classifier_values.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        DB::table('classifier_values')->insert(
+            array_map(function (array $classifierValueData) {
+                return [
+                    ...$classifierValueData,
+                    'type' => ClassifierValueType::FileType->value,
+                    'meta' => '[]',
+                    'created_at' => DB::raw('NOW()'),
+                ];
+            }, $this->getData())
+        );
```
Comment (DanBondarenko) [inline application/database/migrations/2023_04_03_122625_add_file_type_classifier_values.php]: From looking at the code, it seems that the `updated_at` column is not being set. If you use Eloquent ORM to insert data, then the timestamp fields will be set automatically.

Therefore I would suggest something like this (haven’t tested): 
```suggestion
        foreach ($this->getData() as $classifierValueData) {
            ClassifierValue::create([
                ...$classifierValueData,
                'type' => ClassifierValueType::FileType->value,
                'meta' => '[]',
            ]);
        }
```

File: application/database/migrations/2023_04_03_122625_add_file_type_classifier_values.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+return new class extends Migration
+{
+    public function up(): void
+    {
+        DB::table('classifier_values')->insert(
+            array_map(function (array $classifierValueData) {
+                return [
+                    ...$classifierValueData,
+                    'type' => ClassifierValueType::FileType->value,
+                    'meta' => '[]',
```
Comment (DanBondarenko) [inline application/database/migrations/2023_04_03_122625_add_file_type_classifier_values.php]: **[non-blocker]** Hmm, I notice that while the `meta` column is nullable, you are still setting it to an empty array. Not a big deal, but perhaps if the plan is to always default to an empty array, then the column shouldn’t be nullable?

File: application/database/migrations/2023_04_03_122625_add_file_type_classifier_values.php
Comment (DanBondarenko) [inline application/database/migrations/2023_04_03_122625_add_file_type_classifier_values.php:1]: **[non-blocker]** My IDE (PhpStorm) reports that the `DB` class is undefined. It is probably autoloaded by Laravel,  so the code works. But I’m not sure whether this is considered best practice (you can decide). 

In any case, the class could be defined by adding this to the top of the file:
```
use Illuminate\Support\Facades\DB;
```

**Review by DanBondarenko — COMMENTED** [review]
Hi @kadmit, I made some minor suggestions :)

**Comment by kadmit** [comment]
Hi @DanBondarenko! Thank you for the suggestions!

Regarding the usage of Eloquent ORM inside the migrations:
I prefer to do not to use it inside the migrations as it can cause potential issues and unexpected behavior. Using Eloquent inside migrations can lead to inconsistencies between the database schema and the data, which can cause errors.

In case when you change a casting, add a mutation, or add a relationship the model can only work in the present, it can’t work in the past. The right migration behavior, I think, is that they can be applied and reverted and lead to the same result.

**Comment by DanBondarenko** [comment]
> Hi @DanBondarenko! Thank you for the suggestions!
> 
> Regarding the usage of Eloquent ORM inside the migrations: I prefer to do not to use it inside the migrations as it can cause potential issues and unexpected behavior. Using Eloquent inside migrations can lead to inconsistencies between the database schema and the data, which can cause errors.
> 
> In case when you change a casting, add a mutation, or add a relationship the model can only work in the present, it can’t work in the past. The right migration behavior, I think, is that they can be applied and reverted and lead to the same result.

A-ha, I see. Thank you for the explanation! Laravel is new to me. :)

**Review by DanBondarenko — APPROVED** [review]

**Review by thenouan — APPROVED** [review]

**Pull request merged**